client/src/components/Pages/RegElement/Main/RegElement.tsx

```diff
@@ -13,71 +16,93 @@ interface Props {
+  );
+  const [gene, setGene] = useState<string>(
+    regElement?.associated_gene?.label || ""
+  );
+  const [geneText, setGeneText] = useState<string>("");
 
-  // Don't want to change the suggested element. should maybe create a separate context of the
-  // unmutated selected suggestion
-  // TODO -- working
-  // const initialElements = useRef(regElements);
+  /**
+   * Remove regulatory element submitted by user and wipe input fields.
+   */
+  const handleRemove = () => {
+    const cloneFusion = { ...fusion };
+    delete fusion.regulatory_element;
+    setRegElement(undefined);
+    setFusion(cloneFusion);
+    setElementClass("default");
+    setGene("");
+    setGeneText("");
+  };
 
-  const handleRemove = (regEl: ClientRegulatoryElement) => {
-    //copy regulatory elements array, then remove the element with the relevant ID
-    let cloneArray: ClientRegulatoryElement[] = Array.from(regElements);
-    cloneArray = cloneArray.filter((obj) => {
-      return obj["element_id"] !== regEl["element_id"];
-    });
-    setFusion({ ...fusion, ...{ regulatory_elements: cloneArray || [] } });
+  /**
+   * Update state with newly-constructed or updated regulatory element object.
+   * @param element regulatory element constructed from user input by server.
+   */
+  const handleUpdate = (element: ClientRegulatoryElement) => {
+    setRegElement(element);
+    setFusion({ ...fusion, ...{ regulatory_element: element } });
   };
 
   return (
     <div className="reg-element-tab-container">
       <div className="left">
         <div className="blurb-container">
-          {regElements.length > 0 ? (
+          {regElement !== undefined ? (
             <div className="blurb">
               This transcript structure appears to be associated with a
               <p />
-              {regElements.map(
-                (regEl: ClientRegulatoryElement, index: number) => (
-                  <span key={index} className="bold">
-                    {regEl.associated_gene &&
-                      regEl.associated_gene.label &&
-                      regEl.associated_gene.label.toUpperCase()}{" "}
-                    {regEl.hr_class}
-                  </span>
-                )
+              {regElement !== undefined ? (
+                <span key={index} className="bold">
+                  {regElement.associated_gene &&
+                    regElement.associated_gene.label &&
+                    regElement.associated_gene.label.toUpperCase()}{" "}
+                  {regElement.display_class}
+                  {"."}
+                </span>
+              ) : (
+                <></>
               )}
               <p />
-              Regulatory Element.
             </div>
           ) : (
             <div className="blurb">No regulatory element found.</div>
           )}
           <div className="sub-blurb">
-            You can add or remove regulatory elements.
+            {regElement !== undefined
```

Comment:
I think you can remove this check as well and include all of this html in the check on line 57